ape_etherscan/query.py

```diff
@@ -0,0 +1,67 @@
+from typing import Iterator, Optional
+
+from ape.api import QueryAPI, QueryType, ReceiptAPI
+from ape.api.query import AccountTransactionQuery
+from ape.exceptions import QueryEngineError
+from ape.utils import singledispatchmethod
+
+from ape_etherscan.client import ClientFactory
+
+
+class EtherscanQueryEngine(QueryAPI):
+    @property
+    def _client_factory(self) -> ClientFactory:
+        return ClientFactory(
+            self.provider.network.ecosystem.name,
+            self.provider.network.name.replace("-fork", ""),
+        )
+
+    @singledispatchmethod
+    def estimate_query(self, query: QueryType) -> Optional[int]:  # type: ignore[override]
+        return None
+
+    @estimate_query.register
+    def estimate_account_transaction_query(self, query: AccountTransactionQuery) -> int:
+        # About 15 ms per page of 100 transactions
+        return 1500 * (1 + query.stop_nonce - query.start_nonce) // 100
+
+    @singledispatchmethod
+    def perform_query(self, query: QueryType) -> Iterator:  # type: ignore[override]
+        raise QueryEngineError(
+            f"{self.__class__.__name__} cannot handle {query.__class__.__name__} queries."
+        )
+
+    @perform_query.register
+    def get_account_transactions(self, query: AccountTransactionQuery) -> Iterator[ReceiptAPI]:
+        client = self._client_factory.get_account_client(query.account)
+        chain_id = self.provider.chain_id  # TODO: Cache this somehow [APE-635]
+        for receipt_data in client.get_all_normal_transactions():
+            if "confirmations" in receipt_data:
+                receipt_data["required_confirmations"] = receipt_data.pop("confirmations")
+            if "txreceipt_status" in receipt_data:
+                # NOTE: Etherscan uses `""` for `0` in the receipt status.
+                status = receipt_data.pop("txreceipt_status") or 0
+                receipt_data["status"] = status
+
+            if receipt_data.get("nonce") == "":
```

Comment:
A test proving this receipt translation logic is accurate would be nice.

I believe we have mock responses for `get_all_normal_transactions()` already, but we could adjust for a more rich test.

Then we could assert we arrive at `Receipt` objects with expected properties.